### PR TITLE
Proposed changes to simplify custom Field Type creation

### PR DESCRIPTION
**Pull request by kore**
Proposed changes to simplify custom Field Type creation

### CODE REVIEW

File: doc/specifications/proposed/field_type.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
+Field Type Creation Simplification
+==================================
+
+Overview
+--------
+
+First provide a couple of default implementations on the
+`eZ\Publish\Core\FieldType\FieldType` class. Omit all optional methods
+from the
+[tutorial](https://doc.ez.no/display/EZP/eZ+Publish+5+Field+Type+Tutorial).
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: If we omit optional methods, there will be no easy reference on how to implement them. Maybe we should create at least one "advanced" implementation that shows optional methods?

Side note: current version of FT tutorial is [here](https://github.com/ezsystems/TweetFieldTypeBundle/blob/documentation/docs/index.md)

File: doc/specifications/proposed/field_type.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+- Add field type settings
+- Implement a custom value object
+  - Implement custom hashing
+- Enhance the indexing of the field type
+- Store or retrieve external data
+
+This way the developer quickly gets to a functional working result which
+then can be extended. If a new developer has to touch too many classes
+or other code artifacts (configuration …) it will be really hard for
+them to debug all those common subtile issues (naming, …).
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: Typo: subtle (?)

File: doc/specifications/proposed/field_type.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  Internal normalization probably isn't an option since a developer
+  would have to use different identifiers in Twig template blocks then
+  the one specified in the field type implementation. We could assert
+  on a proper identifier in the test case. The documentation suggests
+  to use `.` as a separator in the name – we should ensure this works
+  fine with Twig since I guess this could be considered an operator by
+  Twig, too.
+
+We should probably document the field type base test class
+`eZ\Publish\Core\FieldType\Tests\FieldTypeTest` which allows to test
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: The problem with `Core` namespace is that we tend to change its "contents" without a notice. If we intend for developers to extend it, maybe we should move it to `SPI`?

File: doc/specifications/proposed/field_type.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        return intval($row['contentclass_id']);
+    }
+
+This does not solve the set up issues:
+
+- A non existing table must be created
+- An external database / web service might need to be reset during
+  tests
+
+But since the schema management inside eZ Publish is yet to be defined
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: `eZ Platform`, right? :)

File: doc/specifications/proposed/field_type.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+existing tests:
+
+    bin/phpunit --stop-on-failure path/to/my/FieldTypeIntegrationTest.php
+
+This should be mentioned together with registering the new custom
+storage, which is also is yet undocumented. To make sure this also works
+we can replace the `getCustomHandler()` method in the
+`BaseIntegrationTest` with access to the actual Dependency Injection
+Container to make sure this integration also works. Maybe also add a
+test explicitely testing that the returned handler is correct and
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: Typo: explicitly

File: doc/specifications/proposed/field_type.md [hunk 6]
```diff
@@ -0,0 +1,308 @@
+Field Type Creation Simplification
+==================================
+
+Overview
+--------
+
+First provide a couple of default implementations on the
+`eZ\Publish\Core\FieldType\FieldType` class. Omit all optional methods
```
Comment (alongosz) [inline doc/specifications/proposed/field_type.md]: We should rather encourage implementing `eZ\Publish\SPI\FieldType\FieldType` as there is a BC promise on SPI.

**Review by alongosz — CHANGES_REQUESTED** [review]
Looks good, I found some minor issues:

**Comment by andrerom** [comment]
@alongosz Anything here we should consider to apply to 3.x or 4.x?
Asking as engineering would need to drive this if something here is still relevant.

**Comment by alongosz** [comment]
@andrerom I think this was approached differently and resolved via #2632.

**Pull request closed without merging**